src/Workspaces/VisualBasic/Portable/Extensions/SyntaxNodeExtensions.vb

```diff
@@ -797,11 +797,51 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
         End Function
 
         <Extension>
-        Public Function IsLeftSideOfAnyAssignStatement(node As SyntaxNode) As Boolean
+        Public Function IsLeftSideOfSimpleAssignmentStatement(node As SyntaxNode) As Boolean
             Return node.IsParentKind(SyntaxKind.SimpleAssignmentStatement) AndAlso
                 DirectCast(node.Parent, AssignmentStatementSyntax).Left Is node
         End Function
 
+        <Extension>
+        Public Function IsLeftSideOfAnyAssignmentStatement(node As SyntaxNode) As Boolean
+            Return node IsNot Nothing AndAlso
+                node.Parent.IsAnyAssignmentStatement() AndAlso
+                DirectCast(node.Parent, AssignmentStatementSyntax).Left Is node
+        End Function
+
+        <Extension>
+        Public Function IsAnyAssignmentStatement(node As SyntaxNode) As Boolean
+            Return node IsNot Nothing AndAlso
+                SyntaxFacts.IsAssignmentStatement(node.Kind)
+        End Function
+
+        <Extension>
+        Public Function IsLeftSideOfCompoundAssignmentStatement(node As SyntaxNode) As Boolean
+            Return node IsNot Nothing AndAlso
+                node.Parent.IsCompoundAssignmentStatement() AndAlso
+                DirectCast(node.Parent, AssignmentStatementSyntax).Left Is node
+        End Function
+
+        <Extension>
+        Public Function IsCompoundAssignmentStatement(node As SyntaxNode) As Boolean
+            If node IsNot Nothing Then
+                Select Case node.Kind
```

Comment:
i thnk you can do `select cast node?.Kind()`.  or does that not work in vb?

Reply:
I am not sure, this felt more clear to me :-)

Reply:
works for me!